src/mlte/spec/spec_validator.py

```diff
@@ -32,79 +31,52 @@ def __init__(self, spec: Spec):
+            )
+        self.values[value.metadata.get_id()] = value
 
-    def validate(self) -> BoundSpec:
+    def validate(self) -> ValidatedSpec:
         """
-        Validates the internal properties given its conditions and the stored values, and generates a BoundSpec from it.
+        Validates the internal properties given its requirements and the stored values, and generates a ValidatedSpec from it.
 
         :return: The validated specification
-        :rtype: BoundSpec
-        """
-        results = self._validate_properties()
-        return self.spec.generate_bound_spec(results)
-
-    def _validate_properties(self) -> dict[str, dict[str, Result]]:
+        :rtype: ValidatedSpec
         """
-        Validates a set of conditions by property.
-
-        :return: A document indicating, for each property, a dictionary with the Result for each Condition label.
-        :rtype: dict[str, dict[str, Result]]
-        """
-        # Check that all properties have values to be validated.
-        for property in self.spec.properties:
-            if property.name not in self.values:
-                raise RuntimeError(
-                    f"Property '{property.name}' does not have a value that can be validated."
-                )
+        results = self._validate_results()
+        return ValidatedSpec(self.spec, results)
 
-        # Validate and aggregate the results for all properties.
-        results = {
-            property.name: self._validate_property(
-                property, self.values[property.name]
-            )
-            for property in self.spec.properties
-        }
-        return results
-
-    def _validate_property(
-        self, property: Property, values: dict[str, Value]
-    ) -> dict[str, Result]:
+    def _validate_results(self) -> dict[str, Result]:
         """
-        Validates all conditions for a given property, for the given values.
+        Validates a set of stored results.
 
-        :param property: The property we want to validate.
-        :type property: Property
-
-        :param values: A list of values to validate.
-        :type values: dict[str, Value]
-
-        :return: A dict of Results with the validations for each conditions for this property.
-        :rtype: list[Result]
+        :return: A document indicating the Result for each identifier.
+        :rtype: dict[str, Result]
         """
-        # Check that all conditions have values to be validated.
-        conditions = self.spec.conditions[property.name]
-        for condition in conditions:
-            if condition.label not in values:
-                raise RuntimeError(
-                    f"Condition '{condition.label}' does not have a value that can be validated."
+        # Check that all requirements have values to be validated.
+        for _, requirement_list in self.spec.requirements.items():
+            for requirement in requirement_list:
+                if str(requirement.identifier) not in self.values:
```

Comment:
This has always been an open question for me: do we allow users to validate with insufficient evidence? Do we generate a report that shows that certain properties are not covered by the values collected? I am not sure which is desirable from a user perspective.

I have also considered enabling the above functionality, but then also included a `strict` flag on the validation process that fails immediately if there is insufficient evidence.

Reply:
That is a very good question. I am being more strict now by default, but I am not sure what the best option is. Maybe the strict flag would be a good compromise. Created #149 for this.

Reply:
Great, thanks!